subprojects/core/src/main/java/org/gradle/execution/taskgraph/DefaultTaskExecutionPlan.java

```diff
@@ -401,7 +401,13 @@ public void getNodeValues(TaskInfo node, Collection<? super Void> values, Collec
             }
         });
         graphWalker.add(entryTasks);
-        final List<TaskInfo> firstCycle = new ArrayList<TaskInfo>(graphWalker.findCycles().get(0));
+        List<Set<TaskInfo>> cycles = graphWalker.findCycles();
```

Comment:
When we enter this method it looks like that we should have a cycle. Any ideas why this fails? Isn't there another bug hidden by this exception?